File: frontend/src/types/recommendation.ts
export interface Recommendation {
  rank: number;
  name: string;
  cuisine: string;
  rating: number;
  estimated_cost: string;
  explanation: string;
  restaurant_id?: string | null;
}

export interface RecommendationMetadata {
  total_candidates: number;
  filters_applied: Record<string, unknown>;
  llm_used: boolean;
  filters_relaxed?: string[];
  llm_latency_ms?: number | null;
  token_usage?: Record<string, number> | null;
}

export interface RecommendationResponse {
  summary: string | null;
  recommendations: Recommendation[];
  metadata: RecommendationMetadata;
}

export interface ApiErrorBody {
  detail?: string | { message?: string; suggestions?: string[] };
}

File: app/models/recommendation.py
"""Recommendation response models."""

from __future__ import annotations

from dataclasses import dataclass, field


@dataclass(frozen=True)
class Recommendation:
    rank: int
    name: str
    cuisine: str
    rating: float
    estimated_cost: str
    explanation: str
    restaurant_id: str | None = None


@dataclass(frozen=True)
class RecommendationMetadata:
    total_candidates: int
    filters_applied: dict[str, object]
    llm_used: bool
    filters_relaxed: list[str] = field(default_factory=list)
    llm_latency_ms: float | None = None
    token_usage: dict[str, int] | None = None


@dataclass(frozen=True)
class RecommendationResponse:
    summary: str | None
    recommendations: list[Recommendation]
    metadata: RecommendationMetadata
